applications/tari_merge_mining_proxy/src/proxy.rs

```diff
@@ -134,6 +140,7 @@ struct InnerService {
     config: MergeMiningProxyConfig,
     block_templates: BlockTemplateRepository,
     http_client: reqwest::Client,
+    initial_sync_achieved: Arc<AtomicBool>,
```

Comment:
I don't think that this value needs to be stored. The base node provides it when called, and it doesn't need to be stored and used without calling the base node. Can't it be read from the base node response each time?

Reply:
Yes, it is possible to add the status to `async fn get_new_block_template` instead of to `async fn get_tip_info`, then the code can be simplified as suggested.

Reply:
@mikethetike, this suggestion without having access to the `Arc<AtomicBool>` will result in a poorer user experience, as it will only be possible to inform the user that _"Initial base node sync not achieved, current height at #{}, waiting..."_ while not synced, and not a once-off message of _"Initial base node sync achieved, continuing"_